### PR TITLE
Make factories

### PR DESCRIPTION
**Pull request by mkrumholz**
## Changes

- Added factory bot gem
- Created factories for our models
- Began implementing some basic factories to create items, merchants, invoices for tests
- Refactored some ActiveRecord queries
- Removed some comments
- Updated display for unit pricing on the merchant invoices show page
- Updated my tests to resource variables, rather than bare strings

## Special review requests

It would be great if you could double check me on my refactoring and generally on what I've changed here, since it's a lot of small adjustments. I haven't used all the factories and so can't confirm they're all functional yet, but I figured I'd leave all the outlines in, in case anyone wanted to play with them. 

## Checklist
- [ ] Tests written where required (Factories not tested except where used in views -- not all factories are in use)
- [x] All tests passing
- [x] SimpleCov at 100%

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -71,6 +71,11 @@ GEM
     docile (1.4.0)
     erubi (1.10.0)
     execjs (2.8.1)
+    factory_bot (6.2.0)
+      activesupport (>= 5.0.0)
+    factory_bot_rails (6.2.0)
+      factory_bot (~> 6.2.0)
+      railties (>= 5.0.0)
```
Comment (LawrenceWhalen) [inline Gemfile.lock:78]: So many moving parts.

File: spec/factories/customer_factory.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
+  factory :customer do
+    sequence(:id) { |n| n}
+    first_name { 'Audrey' }
```
Comment (LawrenceWhalen) [inline spec/factories/customer_factory.rb:4]: Does this mean everyone is an audrey? Truly a better timeline.

File: spec/features/merchants/items/index_spec.rb [hunk 3]
```diff
@@ -37,43 +37,43 @@
...
   end
 
   it 'groups items by status' do
     visit "/merchants/#{@merchant.id}/items"
 
     within "section#enabled" do
       expect(page).to have_content 'Enabled Items'
-      expect(page).to have_content 'Audrey II'
-      expect(page).to have_content 'Bouquet of roses'
-      expect(page).to have_content 'Echevaria'
```
Comment (LawrenceWhalen) [inline spec/features/merchants/items/index_spec.rb:75]: The loss of these shops is sad.

**Review by LawrenceWhalen — APPROVED** [review]
Thank you very much for making these. I'll read over the factory bot gem so I can add some more meaningful content.

File: app/models/invoice.rb [hunk 4]
```diff
@@ -29,9 +28,8 @@ def total_revenue
   end
 
   def total_revenue_for_merchant(merchant_id)
-    invoice_items
-    .joins(:item)
-    .where(items: {merchant_id: merchant_id})
+    items
```
Comment (Sidarth20) [inline app/models/invoice.rb:31]: Interesting, so you have access to items in the invoice model by virtue of the joins table?

**Review by Sidarth20 — APPROVED** [review]

File: app/models/invoice.rb (hunk 4 above)
Comment (mkrumholz) [reply app/models/invoice.rb:31]: Yep! Super cool

File: spec/features/merchants/items/edit_spec.rb [hunk 5]
```diff
@@ -15,9 +15,9 @@
         name: 'little-esty-shop'
     })
 
-    @merchant = Merchant.create!(name: "Little Shop of Horrors")
-    @item_1 = @merchant.items.create!(name: 'Audrey II', description: 'Large, man-eating plant', unit_price: '100000000')
-    
+    @merchant = FactoryBot.create(:merchant_with_items)
```
Comment (aliyamerali) [inline spec/features/merchants/items/edit_spec.rb:18]: This is awesome to see in action - thank you for implementing it!

**Review by aliyamerali — APPROVED** [review]
These all look great to me, very cool to see factory bot up and running! And the ActiveRecord refactors are looking good.

**Pull request merged**